### DISCUSSION TITLE
Parsing with line number

### DISCUSSION THREAD

**Discussion by gee4vee** [eemeli/yaml#376] [Q&A] [answered]
I'm looking for a way to parse that would also get me line number information for a specific area in the YAML. Would the Documents or CST parsers allow that? To give an example, I'm working on a validator for a specific kind of YAML and I'd like to give the location of the problem along with information about the problem.

**Comment by eemeli** [comment, marked as answer]
Look for "linecounter" in the docs: https://eemeli.org/yaml/

**Reply by gee4vee** [reply]
Thanks!

**Comment by gee4vee** [comment]
@eemeli do you have any examples of using `LineCounter`? I see it'll give me line and column based on an offset, but how would I know the offset for a particular section of the doc? Do I need to use `Parser.parse` for that?

**Reply by eemeli** [reply]
CST nodes provide their `offset` as an integer, while AST nodes contain `range: [start, value-end, node-end]` positions.

**Comment by the-ricker** [comment]
Here is how I solved the problem
```typescript
let lineCounter = new yaml.LineCounter();
let docs = yaml.parseAllDocuments(Buffer.from(bytes.buffer).toString(), { keepSourceTokens: true, lineCounter: lineCounter});
let api = doc.get('apiVersion') as yaml.Scalar;
let line = lineCounter.linePos(api.range![1]);
console.log("API is " + api.value + " on line " + line[0]);
```
